lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -22,24 +22,29 @@
 #include "clang/Tooling/DependencyScanning/DependencyScanningTool.h"
 #include "llvm/Support/FileSystem.h"
 #include "llvm/Support/Signals.h"
+#include "llvm/Support/Path.h"
 
 using namespace swift;
 
 using namespace clang::tooling;
 using namespace clang::tooling::dependencies;
 
-static std::string lookupModuleOutput(const ModuleID &MID,
-                                      ModuleOutputKind MOK) {
-  // Deciding the output paths is done in swift-driver.
+static std::string moduleCacheRelativeLookupModuleOutput(const ModuleID &MID,
+                                                         ModuleOutputKind MOK,
+                                                         const std::string &moduleCachePathStr) {
+  llvm::SmallString<128> outputPath(moduleCachePathStr);
   switch (MOK) {
   case ModuleOutputKind::ModuleFile:
-    return "<replace-me>";
+    llvm::sys::path::append(outputPath, MID.ModuleName + "-" + MID.ContextHash + ".pcm");
```

Comment:
Could we hoist this logic before the `switch` and then `replace_extension` in non-`ModuleOutputKind::ModuleFile` cases?